src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -89,9 +104,18 @@ class AppsResource @Inject() (
     async {
       implicit val identity = await(authenticatedAsync(req))
 
-      val rawApp = Raml.fromRaml(Json.parse(body).as[raml.App].normalize)
+      val rawApp = Raml.fromRaml(Json.parse(body).as[raml.App].normalize(validateAndNormalizeApp))
       val now = clock.now()
-      val app = validateOrThrow(rawApp).copy(versionInfo = VersionInfo.OnlyVersion(now))
+
+      // This is not really thread safe, another thread may intercept us and change the enforceRole flag, so we need
+      // to revalidate this inside the the groupManager later
+      val roleEnforcement = getEnforcedRoleForService(rawApp.id)
+      val appDefinitionValidator = AppDefinition.validAppDefinition(config.availableFeatures, roleEnforcement)(pluginManager)
+
+      // TODO AN: This should be somewhere else... Normalization maybe?
+      val appWithRole = if (rawApp.role.isDefined) rawApp else rawApp.copy(role = Some(roleEnforcement.defaultRole))
```

Comment:
I don't really understand this line. This results from role being optional I guess. See my comment above.

Reply:
Yeah. It basically sets the appropriate default based on the enclosing group's setting.

Reply:
This is now in the normalization